addons/crm/data/ir_action_data.xml

```diff
@@ -10,7 +10,12 @@
         <field name="binding_model_id" ref="crm.model_crm_lead"/>
         <field name="binding_view_types">list</field>
         <field name="state">code</field>
-        <field name="code">action = model.action_set_lost_with_reason()</field>
+        <field name="code">
+if not 'opportunity' in record.mapped('type'):
+    action = model.action_set_lost()
```

Comment:
We expect these to be called with the relevant recordset. So `records.action_set_lost()`